### ISSUE TITLE
GS 3.3.13/ Apache 2.4 without mod_access_compat - root .htaccess FilesMatch directive causes 500 - Internal Error

### ISSUE THREAD

**Issue by webketje** [GetSimpleCMS/GetSimpleCMS#1207]
Lines 24-25 in `temp.htaccess` cause an internal server error when `mod_access_compat.c` is not enabled because Apache 2.4 doesn't recognize the *Order* and *Deny* statements.
(I've started the server first without, and then with mod_access_compat, in the first case 500 error, in the second it worked).

If I'm not missing something, the whole `<FilesMatch>` directive could be simplified to:

````
<FilesMatch "(?<!sitemap|browserconfig|crossdomain)\.xml$">
  # Apache 2.2
  <IfModule !mod_authz_core.c>
      Order allow,deny
      Deny from all
  </IfModule>
  # Apache 2.4
  <IfModule mod_authz_core.c>
    Require all denied
  </IfModule>	
</FilesMatch>
````

Tested on both Apache 2.2/ PHP 5.4.3 vs. Apache 2.4/ PHP 5.6 and worked in both cases

More info on the subject:

- [Apache.org | Access Control backwards compatibility]( https://httpd.apache.org/docs/2.4/howto/auth.html#filesystem)
- [Apache.org | Upgrading from 2.2 to 2.4](https://httpd.apache.org/docs/trunk/upgrading.html)
- [Hybrid Apache 2.2 - 2.4 .htaccess](https://blog.plee.me/2013/07/access-control-via-hybrid-htaccess-for-both-apache-httpd-2-2-and-2-4/)

**Comment by tablatronix** [comment]
Must be an version inconsistency because I tested this.
Ill take a look can you test the other htaccess files as well?

**Comment by tablatronix** [comment]
See #682

**Comment by webketje** [comment]
I imagine most Apache 2.4 hosts will have the mod_access_compat enabled so this might be an edge case.
I'll test the other htaccess files this evening

**Comment by tablatronix** [comment]
I think I broke it when i added #879 cause I know I tested it under multiple scenarios at first.
Fix sounds reasonable, as long as the backward compatible stuff still works for plugin htaccess files, If you do not have compat all those will fail also, since many have not been updated.

**Comment by webketje** [comment]
Tested the plugins .htaccess on Apache 2.2/2.4, works well. Just had a look at some other flat-file CMS'es .htaccess files for inspiration. Most of them (90%) seem to just use `mod_rewrite` (eg. [Grav](https://github.com/getgrav/grav/blob/develop/.htaccess)). [Bolt](https://github.com/bolt/bolt/blob/release/3.2/.htaccess) has the mod_authz directive as I suggested here, but in inverse order (first 2.4 directive, then 2.2).

**Comment by tablatronix** [comment]
Yeah I added the extra to handle plugins not able to override requires. 

Also avoided rewrites cause you can crash the rewrite engine and bypass security . I do not think rewrites should be used for security

**Comment by webketje** [comment]
Update: same 500 error applies on Apache 2.4 with mod_access_compat disabled for the `/plugins/.htaccess`  file. It works when wrapping the 'Deny from all' statement in an `<IfModule !mod_authz_core.c>`

**Comment by tablatronix** [comment]
I decided against mod_version because it was not guaranteed to be installed and was specifically missing from certain installs, I do not recall exactly, but support for it was less.

**Comment by tablatronix** [comment]
Are you testing stable or master ? cause you posted code from master, but the title says 3.3.13

**Comment by tablatronix** [comment]
These are the rules

```
#
# GetSimple CMS htaccess DENY file
#

# prevent breaking plugin htaccess, prefer compat,  since require is not overridable by order

# apache < 2.3
<IfModule !mod_authz_core.c>
	Deny from all
</IfModule>

# apache > 2.3 with mod_access_compat
<IfModule mod_access_compat.c>
	Deny from all
</IfModule>

# apache > 2.3 without mod_access_compat
<IfModule mod_authz_core.c>

	<IfModule !mod_access_compat.c>
		Require all denied
	</IfModule>

</IfModule>	
```

**Comment by tablatronix** [comment]
Submitted fix for master breaking change #879 

I fail to see a problem in stable
**If a user has 2.4 and does not have mod compat, then they MUST fix their plugin's htaccess or enable it.**
I see no other failure mode in stable.

**Comment by webketje** [comment]
@tablatronix yeah sry didn't mention but it was master indeed. I think I must've read the version from configuration.php.
